### ISSUE TITLE
Error in CDVBMSClient: BMSClient cannot use inherited methods from MFPClient.

### ISSUE THREAD

**Issue by rvplauborg** [ibm-bluemix-mobile-services/bms-clientsdk-cordova-plugin-core#17]
Hi

I believe there is some mistake in the code, considering that `BMSClient`is a subclass of `MFPClient`which has the functions `registerAuthenticationListener`and `unregisterAuthenticationListener`, but when you call `BMSClient.getInstance()`, and get a `BMSClient`object, you cannot call any of the inherited functions..

Example from the CDVBMSClient:
![image](https://cloud.githubusercontent.com/assets/10740421/17810293/8d8ab9dc-661c-11e6-9a5f-bed51789de6f.png)

So in conclusion, I cannot even run my code.

**Comment by rvplauborg** [comment]
So in conclusion, I cannot even run my code.

**Comment by rotembr** [comment]
Hi,
I believe that from some reason Gradle imported Android core version 2.
please check which version of core you are using by switching to Project view:
<img width="406" alt="screen shot 2016-08-21 at 11 09 41 am" src="https://cloud.githubusercontent.com/assets/13929269/17836123/114c1652-6790-11e6-92ac-7beea7e46b68.png">

Also can you specify how you installed the core plugin?

**Comment by rvplauborg** [comment]
Yes it seemed to be version 2.1.1. I installed via the command prompt "cordova plugin add ibm-mfp-core"

**Comment by rotembr** [comment]
Hi, 
Do you added any configuration to the Gradle file? 
What authentication you are trying to configure (google/facebook/custom)?

**Comment by rvplauborg** [comment]
I tried with both Google and Facebook. And I followed a guide: https://console.ng.bluemix.net/docs/services/mobileaccess/facebook-auth-cordova.html 

But i guess I may have added 2.+ in the gradle configuration as the guide tells me to. But what I don't get is that I followed the IBM guide i linked to above completely, and yet it does not work :)

Here is the configuration added: 

`dependencies {
     compile group: 'com.ibm.mobilefirstplatform.clientsdk.android',    
     name:'facebookauthentication',
     version: '2.+',
     ext: 'aar',
     transitive: true
     // other dependencies  
 }`

**Comment by rotembr** [comment]
Please change it to 1.+ (instead of 2.+) in the Gradle file, for example with google:

```
compile group: 'com.ibm.mobilefirstplatform.clientsdk.android',
            name:'googleauthentication',
            version: '1.+',
            ext: 'aar',
            transitive: true 
```

We are working to fix this doc.

**Comment by rvplauborg** [comment]
Alright, done that now. 

According to

> The only difference when you are configuring Cordova applications is that you must initialize the Mobile Client Access client SDK in your JavaScript code instead of in the Java code (see Testing the Authentication). The FacebookAuthenticationManager API must still be registered in your native code

There is no `register`method on the FacebookAuthenticationManager API in 1.+ though - should i just call `FacebookAuthenticationManager.getInstance().registerDefaultAuthenticationListener(this);` instead of `FacebookAuthenticationManager.getInstance().register(this);` then?

As of now, it does not seem to work, although I can build the project now. The Ping test on the app shows "Bummer - unauthorized client"

**Comment by rotembr** [comment]
Yes , that is current.
(updated doc will be available tomorrow)

**Comment by rvplauborg** [comment]
Thank you! I think it is working now actually!
